### PR TITLE
Metadata for snapshots for active active

### PR DESCRIPTION
**Pull request by chbatey**
https://github.com/akka/akka/issues/29286

I hummed and harred and finally went with adding it to the SnapshotMetadata class rather than wrapping so had to do some case class dancing

### CODE REVIEW

File: akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ActiveActiveSnapshotSpec.scala [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        snapshotTestKit.expectNextPersisted(persistenceIdR2, State(List("r1 2", "r1 1")))
+
+        r2.asInstanceOf[ActorRef[Any]] ! internal.PublishedEventImpl(
+          PersistenceId.replicatedUniqueId(entityId, R1),
+          1L,
+          "two-again",
+          System.currentTimeMillis(),
+          Some(new ReplicatedPublishedEventMetaData(R1, VersionVector.empty)))
+
+        // r2 should now filter rout that event if it receives it again
```
Comment (johanandren) [inline akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ActiveActiveSnapshotSpec.scala:90]: filter rout

File: akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ActiveActiveSnapshotSpec.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          PersistenceId.replicatedUniqueId(entityId, R1),
+          1L,
+          "two-again",
+          System.currentTimeMillis(),
+          Some(new ReplicatedPublishedEventMetaData(R1, VersionVector.empty)))
+        r2EventProbe.expectNoMessage()
+      }
+
+    }
+  }
```
Comment (johanandren) [inline akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/ActiveActiveSnapshotSpec.scala:107]: No verification that it actually writes a snapshot, replays from snapshot etc?

**Review by johanandren — COMMENTED** [review]
Mostly looks good, left some comments

**Comment by johanandren** [comment]
PLS BUILD

File: without-gossip.txt [hunk 3]
```diff
@@ -0,0 +1,2975 @@
+JVM-2[0m] 1.287: [GC (Allocation Failure) [PSYoungGen: 65536K->4721K(76288K)] 65536K->4737K(251392K), 0.0117946 secs] [Times: user=0.02 sys=0.00, real=0.01 secs] [0m
```
Comment (johanandren) [inline without-gossip.txt:1]: something unrelated?

**Review by johanandren — APPROVED** [review]
LGTM after 2 unrelated files gone

File: without-gossip.txt (hunk 3 above)
Comment (chbatey) [reply without-gossip.txt:1]: lol oops

**Pull request merged**